### ISSUE TITLE
getPass not working on Positron

### ISSUE THREAD

**Issue by hturner** [wrathematics/getPass#19]
``` r
.Platform$GUI
#> [1] "X11"
Sys.info()["sysname"]
#>  sysname 
#> "Darwin"
```

Positron info:

Positron Version: 2024.09.0 (Universal) build 27
Code - OSS Version: 1.92.0
Commit: d996153f3be6bcc9af460300e61103425323b973
Date: 2024-09-11T02:38:46.408Z
Electron: 30.1.2
Chromium: 124.0.6367.243
Node.js: 20.14.0
V8: 12.4.254.20-electron.0
OS: Darwin arm64 23.6.0

When I run `getPass::getPass()` in Positron, it uses the masked terminal interface. The prompt appears in the console ("PASSWORD:"), but if I type, the text is not masked and pressing enter does not return to the R prompt. I can only get back to the R prompt by killing the command, then the text I have typed appears as if I have typed it at the R prompt.

I didn't see this discussed here or on the Positron repo, let me know if I should raise it with the Positron team instead/as well.

**Comment by idavydov** [comment]
A related positron issue: https://github.com/posit-dev/positron/issues/7978

**Comment by juliasilge** [comment]
I think this line needs to be updated to also return true for Positron:

https://github.com/wrathematics/getPass/blob/a85297e8d7a3b9ed7a07ffa071364b0545a45df2/R/getPass.r#L59

Once that happens, then the function [in this file](https://github.com/wrathematics/getPass/blob/master/R/readline_masked_rstudio.r) should work as expected, because we _do_ support `rstudioapi::askForPassword)` (i.e. `rstudioapi::hasFun("askForPassword")` returns true). I would recommend calling `Sys.getenv("POSITRON")` to check, or actually perhaps `rstudioapi::hasFun("askForPassword")` would be the best guard there, for any environments that support it.

Let me know if you have any questions on this!

**Comment by idavydov** [comment]
It's probably not enough just to change that line.

```
> getPass:::readline_masked_rstudio(msg="prompt", forcemask=F)
NOTE: Masked input is not supported in your version of RStudio; please update to version >= 0.99.879

prompt
password
Error in `readline_nomask()`:
! argument "noblank" is missing, with no default
Show Traceback
```

`rstudioapi::hasFun("askForPassword")` returns `FALSE` when running in Positron. Probably something which needs to be fixed on the positron side, @juliasilge ?

The background is that there are around 20 CRAN packages which reverse depend on `getPass` ☹ 

There's also a tiny bug in `readline_masked_rstudio_window()`, which causes the `argument "noblank" is missing, with no default` message, but that one is only happens in ancient versions of Rstudio. Not sure anyone's using those.

**Comment by idavydov** [comment]
Ah, apologies, in the new version of positron (2025.06.0) that works as expected; so then probably changing that line should be enough.

**Comment by juliasilge** [comment]
Yep, `rstudioapi::hasFun("askForPassword")` should definitely return true in Positron, and I think might be the best thing to check here:

https://github.com/wrathematics/getPass/blob/a85297e8d7a3b9ed7a07ffa071364b0545a45df2/R/getPass.r#L59
